### DISCUSSION TITLE
Sort articles by custom field

### DISCUSSION THREAD

**Discussion by ebbes-isch-emmer** [joomla/joomla-cms#42430] [2. Cold Features]
I would find it very helpful if articles could be sorted by a custom field without having to create an override.
Sort articles by custom field

**Comment by Scrabble96** [comment]
Here I'm copying over my version of this request, which also includes an option to pre-filter by custom field as well as sort:

**Objective:** 
To filter and/or sort by custom field in the back end whilst creating a menu item so that the front-end category blog/category list only displays the pre-filtered articles and/or in custom field sort order

**Current options:** 
When displaying a category blog or list of future events, the only two ways to hide an article using core code after the event date is passed is to -
a. add an 'unpublish' date that matches the event date
b. add an 'unfeature' date if the blog or list only shows featured items.

In order to sort by an event date, when the events may not be added all at the same time, is to manually adjust the publish or feature date so that it slots into the right date order for sorting. That means checking the publish/feature dates of the events immediately before and after the new event, which may not be possible for the person adding the new event. For instance, the other event articles may have been added by another user and, depending on the new user's access level, access to the publishing information may not be accessible by the current user, particularly if the publish date is not displayed on the front end.

Both options (a. and b. above) rely on the person adding the event - who is not necessarily the site admin but quite likely someone non-technie adding the article from the front end - going into the 'Publishing' tab, which may not always be available to front-end editors, and adding the correct 'unpublish' or 'unfeature' date and time. There is no current way (afik) to make these options mandatory.

**Describe the solution you'd like**
Using the events display as an example, if a mandatory custom calendar field (with 'show time') is assigned to the event category then when creating the menu item there would be the option to both:
a. filter the category by the custom calendar field where the value in the custom field is greater than today (or any other chosen date).
b. sort by the custom date (ASC/DESC)

**Additional context:**
This function could apply to other types of field, e.g. a custom email field and to filter where the field is or is not empty.

**Comment by nibra** [comment]
Also see https://github.com/joomla/rfc/blob/cck/proposed/cck-meta.md

**Comment by Scrabble96** [comment]
> Also see https://github.com/joomla/rfc/blob/cck/proposed/cck-meta.md

Thanks, @nibra 
I've had a look at that and, whilst I can see what is proposed and that 12 of 19 voters were in favour of it, I can't see whether the idea is being progressed or whether anything is happening as there is no date on the proposal and no pointers to any code.
